pkg/cvo/sync_worker.go

```diff
@@ -483,6 +491,7 @@ func (w *SyncWorker) syncOnce(ctx context.Context, work *SyncWork, maxWorkers in
 		})
 		info, err := w.retriever.RetrievePayload(ctx, update)
 		if err != nil {
+			w.eventRecorder.Eventf(cvoObjectRef, corev1.EventTypeWarning, "RetrievePayloadFailed", "retrieving payload failed version=%q image=%q failure=%v", update.Version, update.Image, err)
 			reporter.Report(SyncWorkerStatus{
```

Comment:
Seems like we could move the event recorder under `reporter.Report`?

Reply:
> Seems like we could move the event recorder under `reporter.Report`?

that would be a significant change. It's used for all error calls and I'm not certain of the fanout that would cause.  I'd like to solve the immediate "special" events and allow someone closer to the operator decide if and how to emit an event for each report.

Reply:
Yeah, let's keep reporter separate for now.